.github/workflows/main-linter.yml

```diff
@@ -0,0 +1,40 @@
+name: Pylint
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3]
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pylint
+        
+    
+    - name: Analysing the code with pylint
+      run: |
+        pylint --errors-only *.py $(git ls-files '*.py')
+    
+    - name: Format with black
+      run: |
+        pip install black
+        black *.py $(git ls-files '*.py')
```

Comment:
@loganwyas @Massi-Papi 
Not sure you want to format the code as part of this workflow. It would be better to format the code once (maybe @loganwyas can run this 'black' took to do nice formatting) and commit the formatting changes. After that, you can have the linter report formatting problems, so that the team could address them with each commit.